Submitted By: Jim Gifford (jim at linuxfromscratch dot org)
Date: 2003-12-26
Initial Package Version: 2.3.3
Origin: Greg Schafer and Jim Gifford
Description: Fixes Python build issue with gdbm

       *** WARNING: renaming "dbm" since importing it failed:                         
       build/lib.linux-i686-2.3/dbm.so: undefined symbol: dbm_firstkey  


 
$LastChangedBy: igor $
$Date: 2005-01-06 08:31:17 -0700 (Thu, 06 Jan 2005) $

diff -Naur Python-2.3.2.orig/setup.py Python-2.3.2/setup.py
--- Python-2.3.2.orig/setup.py	2003-09-22 01:19:45.000000000 +0000
+++ Python-2.3.2/setup.py	2003-10-13 06:28:34.000000000 +0000
@@ -596,12 +596,12 @@
                 if self.compiler.find_library_file(lib_dirs, 'ndbm'):
                     ndbm_libs = ['ndbm']
                 else:
-                    ndbm_libs = []
+                    ndbm_libs = ['gdbm', 'gdbm_compat']
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_NDBM_H',None)],
                                        libraries = ndbm_libs ) )
             elif (self.compiler.find_library_file(lib_dirs, 'gdbm')
-                  and find_file("gdbm/ndbm.h", inc_dirs, []) is not None):
+                  and find_file("ndbm.h", inc_dirs, []) is not None):
                 exts.append( Extension('dbm', ['dbmmodule.c'],
                                        define_macros=[('HAVE_GDBM_NDBM_H',None)],
                                        libraries = ['gdbm'] ) )
